apps/gallery/js/gallery.js

```diff
@@ -179,42 +179,35 @@ function init() {
   window.performance.mark('navigationInteractive');
   window.dispatchEvent(new CustomEvent('moz-chrome-interactive'));
 
-  // If we were not invoked by an activity, then start off in thumbnail
-  // list mode, and fire up the MediaDB object.
-  if (!navigator.mozHasPendingMessage('activity')) {
-    initDB();
+  // Start initializing the MediaDB
+  initDB();
+
+  // If we were invoked for anything other than a pick activity, we
+  // start off in thumbnail list mode. If we were invoked for a pick
+  // activity, then we need to wait until the activity request actually
+  // arrives below.
+  if (!picking) {
     setView(LAYOUT_MODE.list);
   }
 
-  // Register a handler for activities. This will take care of the rest
-  // of the initialization process.
+  // This function handles pick and browse activity requests
   navigator.mozSetMessageHandler('activity', function activityHandler(a) {
     var activityName = a.source.name;
     switch (activityName) {
     case 'browse':
```

Comment:
All my case labels come out this way, throughout the gallery app code base. That's the way emacs indents switch statements, and I've never tried to figure out how to change it. 

I think it may be an old C convention. The K&R book does it this way. And logically, the entire switch statement is a single block of code with entry points labelled by the cases. Because we can fall through from one block to the next, one could argue (though I'm not sure I would) that double indentation is incorrect because it implies that each case is an independent block.

In any case, my code is littered with this kind of indentation, and I don't think it makes sense to fix just one of them in his patch, especially since the relevant lines aren't actually modified by the patch.